src/main.ts

```diff
@@ -105,6 +112,24 @@ const crawler = new CheerioCrawler({
     additionalMimeTypes: ["application/octet-stream"],
     maxRequestRetries: input.maxRequestRetries,
     requestHandler: router as any,
+    preNavigationHooks: [async () => {
+        const maxLimitReached = scrapeSettings.totalPushedResults >= scrapeSettings.maxResults;
+
+        if (maxLimitReached) {
+            const errorMessage = `Reached limit of max crawled places for this search term, skipping all next requests in the queue for this search `
+                + `(this might take a while, don't mind the errors). [Draining request queue]`;
+            if (!scrapeSettings.startedDrainingState === true) {
+                await Actor.setStatusMessage(errorMessage);
+            }
+
+            // We don't want anything to override our status message
+            // @ts-expect-error - this is a hack to override the status message
+            crawler.setStatusMessage = () => { /* do nothing */ };
+
+            scrapeSettings.startedDrainingState = true
+            throw new NonRetryableError(errorMessage);
+        }
+    }],
```

Comment:
I hope that this shouldn't be necessary, when we teardown the crawler

or is our SDK really _that_ unhelpful?

Reply:
confirmed with `console.log`s: after `crawler.teardown()`, no requests enter the prenavigation hook